submission/pipelines/generate_pipelines.py

```diff
@@ -598,21 +612,31 @@ def main():
     ]
 
     # add our basic pipelines to the submission
-    for (problem_type, problem_name) in [('classification', '185_baseball_MIN_METADATA'), ('regression', '196_autoMpg_MIN_METADATA')] + challenge_problems:
-        is_challenge_prob = problem_name in challenge_names
+    for problem in problems + challenge_problems:
+        is_challenge_prob = problem.name in challenge_names
 
         for primitive_data in primitives_data:
             primitive = primitive_data['primitive']
             # generate and update the pipeline for this primitive
             pipeline_json = generate_and_update_primitive_pipeline(
                 primitive,
                 primitive_data['gen_method'],
-                problem_type,
+                problem.problem_type,
                 is_challenge_prob
             )
-            # save it into the primitives submodule for TA1 submission
+
             primitive_path = primitive.metadata.query()['python_path']
             submission_path = os.path.join(byu_dir, primitive_path, primitive_data['version'])
+            pipeline_run_name = f'{pipeline_json["id"]}_{problem.name}'
+            pipeline_run_path = os.path.join(submission_path, 'pipeline_runs', f"{pipeline_run_name}.yml.gz")
+            if os.path.isfile(pipeline_run_path):
+                print(
+                    f"pipeline {pipeline_json['id']} has already "
+                    f"been run on problem {problem.name}, skipping."
+                )
+                continue
```

Comment:
will this cause problems when we run things multiple times, say if it failed the first time?